File: backend/internal/model/project_update.go
package model

import "time"

type ProjectUpdate struct {
	ID        uint64    `gorm:"primaryKey" json:"id"`
	ProjectID uint64    `gorm:"index" json:"project_id"`
	AuthorID  uint64    `json:"author_id"`
	Author    User      `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
	Status    string    `gorm:"size:20;not null" json:"status"` // on_track, at_risk, off_track
	Content   string    `gorm:"type:text" json:"content"`
	Recap     string    `gorm:"type:text" json:"recap,omitempty"`     // what was accomplished
	Plan      string    `gorm:"type:text" json:"plan,omitempty"`      // what's next
	Blockers  string    `gorm:"type:text" json:"blockers,omitempty"`  // obstacles
	Metrics   string    `gorm:"type:text" json:"metrics,omitempty"`   // JSON string of key metrics
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/api/project-update.ts
import api from './index'

export interface ProjectUpdate {
  id: number; project_id: number; author_id: number
  author?: { id: number; display_name: string; avatar_url?: string }
  status: 'on_track' | 'at_risk' | 'off_track'
  content: string; created_at: string
}

export async function listUpdates(projectId: number, limit = 20): Promise<ProjectUpdate[]> {
  const r = await api.get(`/projects/${projectId}/updates?limit=${limit}`); return r.data.data || []
}

export async function createUpdate(projectId: number, status: string, content: string): Promise<ProjectUpdate> {
  const r = await api.post(`/projects/${projectId}/updates`, { status, content }); return r.data.data
}

export const projectUpdateApi = { list: listUpdates, create: createUpdate }
